include/pagmo/problems/dtlz.hpp

```diff
@@ -325,7 +325,7 @@ class dtlz
         for (decltype(f.size()) i = 0u; i < f.size() - 1; ++i) {
             y += (f[i] / (1.0 + g)) * (1.0 + std::sin(3 * detail::pi() * f[i]));
         }
-        return m_fdim - y;
+        return static_cast<double>(m_fdim) - y;
```

Comment:
I still wonder why I do not get those warnings. I have tried debug and release in clang and gcc in my architectures ... mah